src/main/frontend/db/query_dsl.cljs

```diff
@@ -517,11 +517,18 @@
                                         (take @sample (shuffle col)))
                                       identity)
                      transform-fn (comp sort-by random-samples)]
-                 (react/react-query repo
-                                    {:query query
-                                     :query-string query-string}
-                                    {:use-cache? false
-                                     :transform-fn transform-fn}))))))))))
+                 (try
+                   (react/react-query repo
+                                      {:query query
+                                       :query-string query-string
+                                       :throw-exception true}
+                                      {:use-cache? false
+                                       :transform-fn transform-fn})
+                   (catch ExceptionInfo e
+                     ;; Allow non-existent page queries to be ignored
+                     (if (string/includes? (str (.-message e)) "Nothing found for entity")
```

Comment:
This is to allow these cases to pass - https://github.com/logseq/logseq/blob/174f3d40a8208c761a5a14e22420eaa3c3d5b98d/src/test/frontend/db/query_dsl_test.cljs#L199-L202 . A better behavior than this would be to notify the user that the entity doesn't exist. I didn't change it as that requires more thought